utils/build-script-impl

```diff
@@ -1398,7 +1398,7 @@ function calculate_targets_for_host() {
            [[ "${build_external_benchmark_this_target}" ]] &&
            [[ "${is_in_build_list}" ]] ; then
             SWIFT_BENCHMARK_TARGETS+=("swift-benchmark-${stdlib_deployment_target}-external")
-            if [[ $(not ${SKIP_TEST_BENCHMARK}) ]] ; then
```

Comment:
I think this would have been expanding to just `not`, which is equivalent to `false`, and then trying to substitute the result, which is `if [[ "" ]]`, which is `false`. @gparker42, does my shell analysis make sense?

Reply:
I don't understand the relevance.

Reply:
You're changing this code from always producing `false` to actually trying to execute something. That's not your fault because the old code is very wrong, but it explains the failure.

Reply:
Sorry in my head, I had already gone beyond that. It is actually sort of funny, the reason that today we do not add something here is b/c of another hack in build-script that explicitly states that this typo is why we are doing a workaround... too bad we didn't just fix it = p.

Reply:
You're going to have to fix this to get anything to work. LLVM `not` takes a command name, not a value. (I *think* the right code here is `if [[ ! "${SKIP_TEST_BENCHMARKS}" ]]` but I haven't tried it.)

Reply:
This is not invoking LLVM ``not``. This is invoking a bash function called ``not`` defined in build-script-impl that has exactly what you described as its implementation. See:

https://github.com/apple/swift/blob/9aca727949f1338cb04855bca32e540acf998cf0/utils/build-script-impl#L962

Reply:
Beyond that do you have anything else?

Reply:
Oh wow that's confusing. Then my earlier diagnosis of why things were failing was incorrect. Sorry for wasting your time!

Reply:
You didn't waste my time! This is just confusing = p.